managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -2707,6 +2707,26 @@ public void operationFailed(ManagedLedgerException exception) {
         });
     }
 
+    /**
+     * Try set {@link #state} to {@link State#Closing}.
+     * @return false if the {@link #state} already is {@link State#Closing} or {@link State#Closed}.
+     */
+    private boolean trySetStateToClosing() {
+        if (STATE_UPDATER.compareAndSet(this, State.Uninitialized, State.Closing)){
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReferenceFieldUpdater.html#updateAndGet-T-java.util.function.UnaryOperator-

What about using updateAndGet?
This way we can use a switch block

Reply:
Thanks, this is a really good way to deal with this problem. 

already fixed